starlite/app.py

```diff
@@ -164,7 +166,7 @@ def __init__(
         self.compression_config = compression_config
         self.plain_routes: Set[str] = set()
         self.plugins = plugins or []
-        self.route_map: Dict[str, Any] = {}
+        self.route_map: RouteMapDict = {}
```

Comment:
So note on the `RouteMapDict` type - I dont really see the point of using it TBH over using Dict[str, Any]. Its not such a big deal so its fine to include if you feel strongly about it, but I believe in being explicit wherever this doesn't hurt code cleanliness. 

I have a comment lower regarding naming which is relevant.

Reply:
After including the sentinel, the type becomes `Dict[Union[str, PathParamPlaceholderType], Any]` which in some places (like in `cast()` calls) makes the annotation long/dense.